Rahul - PR Review Dash - Top 20 Most Popular PRs

slidracoon72 opened the pull request:
# Description
<img width="614" height="677" alt="Screenshot 2025-07-14 at 12 12 29 PM" src="https://github.com/user-attachments/assets/9306c9d6-2ac3-4604-986a-2de350900a86" />
<img width="585" height="352" alt="image" src="https://github.com/user-attachments/assets/c811dc37-f9ff-4fc5-b6da-6198a02a0c44" />

## Related PRS:
To test this frontend PR, you need to checkout the `development` backend PR.

## Main changes explained:
- Created the horizontal bar graph titled “Top 20 Most Popular PRs”, displaying PR numbers (Y-axis) vs. number of reviews (X-axis) with data labels.
- Added filter dropdown with duration options: Last Week (default), Last 2 Weeks, Last Month, and All Time to dynamically control the data shown.
- Implemented API integration and state updates to fetch and re-render the graph on filter selection change.
- Handled loading and error states with appropriate UI feedback during data fetch.
- Enabled vertical scroll when more than 10 PRs are displayed to ensure clean UI and usability.
- Made layout responsive for smaller screens, including optimized styling and layout adjustments.
- Added tooltips to display additional PR metadata (e.g., title, link) on hover over bars.
- Updated styles to match the Figma design specifications (colors, typography, spacing).
- Added theme context to switch between light and dark modes.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. clear site data/cache
4. log as admin/owner user
5. go to http://localhost:3000/pr-team-analytics/popular-prs 
6. verify the horizontal graph is working as expected
7. verify the dropdown to select different duration options works correctly
8. verify the graph is responsive (for small, medium and large screens)
9. verify the tooltips displays the correct data for the respective PR
10. verify the vertical scrolling within the graph works as expected
11. verify the above features in dark mode as well

## Screenshots and video of changes:
Video Link:
https://github.com/user-attachments/assets/b3308c9b-3354-4d01-859b-d1f2c0384873

Light Mode:
<img width="1309" height="830" alt="Screenshot 2025-07-17 at 2 51 50 PM" src="https://github.com/user-attachments/assets/dc943a35-210a-46e3-b119-7878f7d19fbb" />

Dark Mode:
<img width="1309" height="830" alt="Screenshot 2025-07-17 at 2 33 16 PM" src="https://github.com/user-attachments/assets/aeab4803-769e-4008-a135-c032eb69091c" />


## Note:
The graph currently uses mock data for PR reviews. It will be updated to use real data once the backend API, which integrates with the GitHub API to fetch actual PR review data, is available.

harshavarma29 reviewed (changes requested):
I have tested the feature in the local environment with the admin account. Currently, with the mock data from the PRData.js file, it is working fine. I have requested a few changes as listed below: 

Request 1: Please refer to the comments added in the PRData.js file.
Request 2: The Chart UI is not getting changed in dark mode, as shown in the screenshot below.
![Screenshot 2025-07-15 163030](https://github.com/user-attachments/assets/92125292-9af2-4d0a-bfc0-cb782e3f4205)

sankargalla reviewed (changes requested):
I have tested the implementation locally using an admin account. The dashboard data is displaying correctly as defined in the PRData.js file.

Observation:
1. PR dashboard is not compatible with dark mode, please refer to below screenshots.

<img width="1440" height="778" alt="Screenshot 2025-07-16 at 4 52 16 PM" src="https://github.com/user-attachments/assets/04702542-8e58-4cc4-a7ef-68b451a1cdde" />

<img width="1440" height="769" alt="Screenshot 2025-07-16 at 4 39 41 PM" src="https://github.com/user-attachments/assets/9a66125a-3845-4b42-823a-8f1c878b8a64" />
<img width="1440" height="777" alt="Screenshot 2025-07-16 at 4 39 17 PM" src="https://github.com/user-attachments/assets/e929e98e-dcf0-464f-b333-03e087dd3182" />

slidracoon72 commented on src/components/HGNPRDashboard/PRData.js:
@harshavarma29 
Added additional PRs (total of 50) to cover more test scenarios. 
Also implemented dark mode support for the graph.

slidracoon72 commented:
Added additional mock PRs (total of 50) to cover more test scenarios.
Also implemented dark mode support for the graph.

<img width="1309" height="830" alt="Screenshot 2025-07-17 at 2 33 16 PM" src="https://github.com/user-attachments/assets/1a98809a-b0ec-4be2-b83f-56feaaabfce8" />

naiduajaykumar reviewed (changes requested):
Everything works as intended expect the black theme when I try to apply dark theme every component got dark theme except the background.
No issues with the screen changes it adapts well in different screens.
<img width="1919" height="1069" alt="Screenshot 2025-07-17 203153" src="https://github.com/user-attachments/assets/b888e5bd-5789-4bad-8deb-fd54360136f9" />
<img width="873" height="996" alt="Screenshot 2025-07-17 203235" src="https://github.com/user-attachments/assets/27139f90-da52-42e7-a055-00334cb1a9cd" />
<img width="1916" height="1063" alt="Screenshot 2025-07-17 203416" src="https://github.com/user-attachments/assets/5571c273-65bf-483a-90a0-d869bfcfa338" />

slidracoon72 commented:
> Everything works as intended expect the black theme when I try to apply dark theme every component got dark theme except the background. No issues with the screen changes it adapts well in different screens. <img alt="Screenshot 2025-07-17 203153" width="1919" height="1069" src="https://private-user-images.githubusercontent.com/83155329/467803699-b888e5bd-5789-4bad-8deb-fd54360136f9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI3OTk1NjQsIm5iZiI6MTc1Mjc5OTI2NCwicGF0aCI6Ii84MzE1NTMyOS80Njc4MDM2OTktYjg4OGU1YmQtNTc4OS00YmFkLThkZWItZmQ1NDM2MDEzNmY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE4VDAwNDEwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMzOGQ4NzAxMDE5MTZmOGM2NTRiNDUyZDUzMmY1OGMxZTA1MjVkNTRhM2Y2ZTUzZmRiZDk1MjljNTE4ODhiMGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.i3q84S6aNhQ8Sh1DDe471ocJtdirxLcRHe0rJ2oF2cU"> <img alt="Screenshot 2025-07-17 203235" width="873" height="996" src="https://private-user-images.githubusercontent.com/83155329/467803700-27139f90-da52-42e7-a055-00334cb1a9cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI3OTk1NjQsIm5iZiI6MTc1Mjc5OTI2NCwicGF0aCI6Ii84MzE1NTMyOS80Njc4MDM3MDAtMjcxMzlmOTAtZGE1Mi00MmU3LWEwNTUtMDAzMzRjYjFhOWNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE4VDAwNDEwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThlZWY0YzU5ZDM0ZGNiNzFiOWI5NWViMzdiOTdkODVmMzk0OWY0ZmMyYjE0ZWZlYWYzYWMwMDEzODgzZTAyNzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6D_WJSyOEP_QdDg6I6Ax4ObbZCChImFsoTnmRuu_zAQ"> <img alt="Screenshot 2025-07-17 203416" width="1916" height="1063" src="https://private-user-images.githubusercontent.com/83155329/467803701-5571c273-65bf-483a-90a0-d869bfcfa338.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI3OTk1NjQsIm5iZiI6MTc1Mjc5OTI2NCwicGF0aCI6Ii84MzE1NTMyOS80Njc4MDM3MDEtNTU3MWMyNzMtNjViZi00ODNhLTkwYTAtZDg2OWJmY2ZhMzM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE4VDAwNDEwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRkMzQ3NGNhMTI1NjQ3YjIyYjU2ZGMzNGQ4YmI0ODc5NzAxYTljZGRiYmJmZWY4NjcyZTZlZjZhNDM1ZjAwYjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Jo8EljYkrm3hDhdvoIIHc_GA_iLWY3FGt0HeR4S6-wM">

The graph background is intentionally kept light in dark mode to enhance visibility and ensure clear contrast with the data elements.

slidracoon72 commented:
@shashankm9 
It’s working correctly on my end. It looks like the version you tested might be outdated. I’ve already made the necessary updates, so it should work on your system as well. Please try pulling the latest version of the PR and testing again.

sreeja013 commented:
Everything is working perfectly, but in small screen resolution, there is a small defect. Code looks good, great work!
<img width="670" height="1001" alt="Screenshot 2025-07-18 205846" src="https://github.com/user-attachments/assets/f5abf840-c9de-460c-b54a-2272832e30f0" />

Rishitha1810 commented:
I navigated to http://localhost:3000/pr-team-analytics/popular-prs and verified that all key features are working as expected. The horizontal graph renders correctly, and the dropdown for selecting different duration options functions properly. I confirmed that the graph is responsive across small, medium, and large screen sizes, tooltips display accurate data for each PR, and vertical scrolling within the graph works smoothly. I also tested all of these features in dark mode to ensure consistent behavior and appearance.
<img width="1470" height="798" alt="Screenshot 2025-07-18 at 11 58 31 PM" src="https://github.com/user-attachments/assets/64b47b7a-e147-4b5a-a5e3-7d205d093ca0" />
<img width="1470" height="798" alt="Screenshot 2025-07-18 at 11 58 48 PM" src="https://github.com/user-attachments/assets/68db8bad-f245-4116-afda-3f99a04793a3" />

ChinmayJoshi98 reviewed (approved):
<img width="718" height="509" alt="3746 1" src="https://github.com/user-attachments/assets/f276e48e-b546-4893-ac80-f43c7de52b80" />

I tested this locally. The top 20 PRs dashboard works well: 
graph renders accurately, duration filters function correctly, tooltips display as expected, scrolling is smooth, and responsiveness is intact. 
Dark mode handling is acceptable given the graph’s light background explanation and the overall theme of the website

Neeraj-Kondaveeti reviewed (approved):
Everything is working as expected. The horizontal bar graph renders correctly, the duration filter updates the data dynamically, tooltips display the correct PR information, and the layout is responsive across screen sizes. Dark mode functionality is also verified. Approving the PR.

<img width="1470" height="873" alt="3746" src="https://github.com/user-attachments/assets/84882d85-7f92-4351-a347-96068feb8a04" />

<img width="1470" height="933" alt="3746(2)" src="https://github.com/user-attachments/assets/480fd9d3-354d-4ed7-bf86-7871ad2169a2" />

gurusaichittoji7 commented:
I checked out the current branch, installed dependencies, cleared cache, and ran the app locally. Logged in as an admin/owner user and navigated to /pr-team-analytics/popular-prs. Verified the horizontal graph functionality, dropdown filter for duration, graph responsiveness across screen sizes, and tooltip accuracy. Also confirmed vertical scrolling works within the graph and ensured all features behave correctly in dark mode. Everything worked as expected.
<img width="1911" height="971" alt="Screenshot 2025-07-22 123139" src="https://github.com/user-attachments/assets/88e9fadf-7a60-4182-9575-6edc23fce475" />
<img width="1901" height="966" alt="Screenshot 2025-07-22 123157" src="https://github.com/user-attachments/assets/30bf47a7-a689-4c56-b204-2c0471a7ac0d" />

rohitmamidi commented:
Dark mode is working well!

In the UI components, the Y-axis is labeled as “PR numbers,” but it currently displays the same value as the title (Top 20 most popular PRs). Please review and update the Y-axis label to reflect the actual PR numbers.

For step 8, the graph is responsive across small, medium, and large screens.
 However, the “Back to top” button at the bottom is overlapping/merging with the bar below. Please adjust its position or spacing.

Other than these points, everything looks good.

<img width="2940" height="1912" alt="Screenshot 2025-07-23 at 12 30 01 PM" src="https://github.com/user-attachments/assets/61ca0e7e-8f1c-4d2b-bfbd-34529515ca1d" />

<img width="671" height="660" alt="Screenshot 2025-07-23 at 12 42 52 PM" src="https://github.com/user-attachments/assets/414090c9-d964-4c47-9941-d8099da3ac2f" />

beblicarl commented:
This functionality works as intended and the code is great

<img width="822" height="546" alt="image" src="https://github.com/user-attachments/assets/dfdc2c49-b7f6-455e-82f2-3868da78b875" />

<img width="823" height="538" alt="image" src="https://github.com/user-attachments/assets/83b24c73-9a17-448b-90bc-71f837b633ea" />

one-community commented:
Thank you all, merging!
